### PR TITLE
WIP: Mention and link to the macOS readme in the repo.

### LINKED ISSUE #453: Setup guide should link to platform readme (macOS)
Today I was reading the devguide to try to understand how framework builds work, but found it inadequate. I later learned that the Mac/README file in the repo has more detail. The devguide should reference that in `setup/#macos`. Filing this ticket so I don't forget.

### PR DESCRIPTION
**Pull request by jaraco**
Fixes #453.

### CODE REVIEW

**Comment by jaraco** [comment]
In the ticket, Terry has asked that we defer this link until the link target is updated. I'll file a ticket for that.

**Comment by jaraco** [comment]
Blocked on bpo-35905.

**Review by lastthyme — APPROVED** [review]

File: setup.rst
Comment (matrixise) [inline setup.rst:409]: Use the new URL with the main branch: https://github.com/python/cpython/blob/main/Mac/README.rst

**Review by matrixise — CHANGES_REQUESTED** [review]

File: setup.rst
Comment (jaraco) [reply setup.rst]: Done

File: setup.rst
Comment (ezio-melotti) [inline setup.rst:408]:
```suggestion
to the `macOS README
```

**Comment by za** [comment]
Thanks @ezio-melotti for merging. We may remove this label: `status-do-not-merge` https://github.com/python/devguide/pull/454#event-2399380486

**Comment by ezio-melotti** [comment]
I thought I commented on this issue, but apparently I forgot to submit after I merged.  The README has been updated since this PR was created and I think a link to the README is useful regardless.  I therefore decided to go ahead and merge it even if https://github.com/python/cpython/issues/80086 is still open.

**Pull request merged**
